File: analisis_basico/condicionales/if_case_10/if_case_10.cpp

#include <iostream>
#include <fstream>
int readRandomBin(int *&array_int, int &num){
    std::ifstream file("../random_list.bin", std::ios::binary);
    if (!file) {
        return 1;
    }

    file.seekg(0, std::ios::end);
    long file_size = file.tellg();
    file.seekg(0, std::ios::beg);

    num = file_size / sizeof(int);
    array_int = new int[num];

    file.read(reinterpret_cast<char*>(array_int), file_size);

    file.close();
    return 0;
}

int main() {
    
    int num;
    int *array_int;

    if (readRandomBin(array_int, num) != 0) {
        std::cerr << "Error al abrir" << std::endl;
        return 1;
    }

    for (int i = 0; i < num; i++){
        int value = array_int[i] % 10;
        if (value == 0) {
        } else if (value == 1) {
        } else if (value == 2) {
        } else if (value == 3) {
        } else if (value == 4) {
        } else if (value == 5) {
        } else if (value == 6) {
        } else if (value == 7) {
        } else if (value == 8) {
        } else if (value == 9) {
        }
    }
    delete[] array_int;
    return 0;
}

File: analisis_basico/condicionales/if_case_10/if_case_10.py

def readRandomBin():
    try:
        with open("../random_list.bin", "rb") as file:
            array_bytes = file.read()
    except:
        print("Error al abrir")
        exit(1)

    num = len(array_bytes) // 4
    array_int = [int.from_bytes(array_bytes[i:i+4], byteorder='little') for i in range(0, len(array_bytes), 4)]

    return num, array_int

if __name__ == "__main__":
    num, array_int = readRandomBin()

    for i in range(num):
        value = array_int[i] % 10
        if value == 0:
            pass
        elif value == 1:
            pass
        elif value == 2:
            pass
        elif value == 3:
            pass
        elif value == 4:
            pass
        elif value == 5:
            pass
        elif value == 6:
            pass
        elif value == 7:
            pass
        elif value == 8:
            pass
        elif value == 9:
            pass
